### ISSUE TITLE
[Failing Test] [sig-cli] Kubectl client [k8s.io] Simple pod should handle in-cluster config

### ISSUE THREAD

**Issue by alejandrox1** [kubernetes/kubernetes#81845]
**Which jobs are failing / job(s) where test is present**:
* ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel
* ci-kubernetes-e2e-gce-new-master-gci-kubectl-skew

**Which test(s) are failing**:
[sig-cli] Kubectl client [k8s.io] Simple pod should handle in-cluster config

**Since when has it been failing**:
It has been failing since 8/13.

**Testgrid link**:
* https://testgrid.k8s.io/sig-release-orphaned#gce-new-master-upgrade-master-parallel
* https://testgrid.k8s.io/sig-cli-master#skew-cluster-stable1-kubectl-latest-gce

**Reason for failure**:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:621
Expected
    <exec.CodeExitError>: {
        Err: {
            s: "error running &{../../../../kubernetes_skew/cluster/kubectl.sh [../../../../kubernetes_skew/cluster/kubectl.sh --server=https://34.83.124.226 --kubeconfig=/workspace/.kube/config exec --namespace=kubectl-5797 nginx -- /bin/sh -x -c /tmp/kubectl get pods --token=invalid --v=7 2>&1] []  <nil> I0821 19:51:49.587176     102 merged_client_builder.go:164] Using in-cluster namespace\nI0821 19:51:49.587839     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.594295     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.602903     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.603669     102 round_trippers.go:420] GET https://10.0.0.1:443/api/v1/namespaces/kubectl-5797/pods?limit=500\nI0821 19:51:49.603894     102 round_trippers.go:427] Request Headers:\nI0821 19:51:49.604071     102 round_trippers.go:431]     Accept: application/json;as=Table;v=v1beta1;g=meta.k8s.io, application/json\nI0821 19:51:49.604204     102 round_trippers.go:431]     User-Agent: kubectl/v1.17.0 (linux/amd64) kubernetes/90df64b\nI0821 19:51:49.604360     102 round_trippers.go:431]     Authorization: Bearer <masked>\nI0821 19:51:49.628462     102 round_trippers.go:446] Response Status: 401 Unauthorized in 23 milliseconds\nI0821 19:51:49.629333     102 helpers.go:199] server response object: [{\n  \"kind\": \"Status\",\n  \"apiVersion\": \"v1\",\n  \"metadata\": {},\n  \"status\": \"Failure\",\n  \"message\": \"Unauthorized\",\n  \"reason\": \"Unauthorized\",\n  \"code\": 401\n}]\nF0821 19:51:49.629526     102 helpers.go:114] error: You must be logged in to the server (Unauthorized)\n + /tmp/kubectl get pods '--token=invalid' '--v=7'\ncommand terminated with exit code 255\n [] <nil> 0xc0048fcf90 exit status 255 <nil> <nil> true [0xc00592a1e8 0xc00592a200 0xc00592a218] [0xc00592a1e8 0xc00592a200 0xc00592a218] [0xc00592a1f8 0xc00592a210] [0x9d21f0 0x9d21f0] 0xc001ba2360 <nil>}:\nCommand stdout:\nI0821 19:51:49.587176     102 merged_client_builder.go:164] Using in-cluster namespace\nI0821 19:51:49.587839     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.594295     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.602903     102 merged_client_builder.go:122] Using in-cluster configuration\nI0821 19:51:49.603669     102 round_trippers.go:420] GET https://10.0.0.1:443/api/v1/namespaces/kubectl-5797/pods?limit=500\nI0821 19:51:49.603894     102 round_trippers.go:427] Request Headers:\nI0821 19:51:49.604071     102 round_trippers.go:431]     Accept: application/json;as=Table;v=v1beta1;g=meta.k8s.io, application/json\nI0821 19:51:49.604204     102 round_trippers.go:431]     User-Agent: kubectl/v1.17.0 (linux/amd64) kubernetes/90df64b\nI0821 19:51:49.604360     102 round_trippers.go:431]     Authorization: Bearer <masked>\nI0821 19:51:49.628462     102 round_trippers.go:446] Response Status: 401 Unauthorized in 23 milliseconds\nI0821 19:51:49.629333     102 helpers.go:199] server response object: [{\n  \"kind\": \"Status\",\n  \"apiVersion\": \"v1\",\n  \"metadata\": {},\n  \"status\": \"Failure\",\n  \"message\": \"Unauthorized\",\n  \"reason\": \"Unauthorized\",\n  \"code\": 401\n}]\nF0821 19:51:49.629526     102 helpers.go:114] error: You must be logged in to the server (Unauthorized)\n\nstderr:\n+ /tmp/kubectl get pods '--token=invalid' '--v=7'\ncommand terminated with exit code 255\n\nerror:\nexit status 255",
        },
        Code: 255,
    }
to contain substring
    <string>: Authorization: Bearer invalid
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:721
```

I think the failure is coming from this part of the test https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L720

/milestone v1.16
/priority critical-urgent
/kind failing-test
/sig sig-cli
/cc @kubernetes/sig-cli-test-failures 
/cc @Verolop @jimangel @soggiest @alenkacz

**Comment by alejandrox1** [comment]
More organized look at the logs - https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-master-parallel/1164253841420455936#0:build-log.txt%3A26115
```
I0821 19:52:04.840] STEP: trying to use kubectl with invalid token
I0821 19:52:04.840] Aug 21 19:51:47.350: INFO: Running '../../../../kubernetes_skew/cluster/kubectl.sh --server=https://34.83.124.226 --kubeconfig=/workspace/.kube/config exec --namespace=kubectl-5797 nginx -- /bin/sh -x -c /tmp/kubectl get pods --token=invalid --v=7 2>&1'
I0821 19:52:04.840] Aug 21 19:51:49.770: INFO: rc: 255
I0821 19:52:04.841] Aug 21 19:51:49.770: INFO: got err error running &{../../../../kubernetes_skew/cluster/kubectl.sh [../../../../kubernetes_skew/cluster/kubectl.sh --server=https://34.83.124.226 --kubeconfig=/workspace/.kube/config exec --namespace=kubectl-5797 nginx -- /bin/sh -x -c /tmp/kubectl get pods --token=invalid --v=7 2>&1] []  <nil> I0821 19:51:49.587176     102 merged_client_builder.go:164] Using in-cluster namespace
I0821 19:52:04.841] I0821 19:51:49.587839     102 merged_client_builder.go:122] Using in-cluster configuration
I0821 19:52:04.841] I0821 19:51:49.594295     102 merged_client_builder.go:122] Using in-cluster configuration
I0821 19:52:04.842] I0821 19:51:49.602903     102 merged_client_builder.go:122] Using in-cluster configuration
I0821 19:52:04.842] I0821 19:51:49.603669     102 round_trippers.go:420] GET https://10.0.0.1:443/api/v1/namespaces/kubectl-5797/pods?limit=500
I0821 19:52:04.842] I0821 19:51:49.603894     102 round_trippers.go:427] Request Headers:
I0821 19:52:04.843] I0821 19:51:49.604071     102 round_trippers.go:431]     Accept: application/json;as=Table;v=v1beta1;g=meta.k8s.io, application/json
I0821 19:52:04.843] I0821 19:51:49.604204     102 round_trippers.go:431]     User-Agent: kubectl/v1.17.0 (linux/amd64) kubernetes/90df64b
I0821 19:52:04.843] I0821 19:51:49.604360     102 round_trippers.go:431]     Authorization: Bearer <masked>
I0821 19:52:04.844] I0821 19:51:49.628462     102 round_trippers.go:446] Response Status: 401 Unauthorized in 23 milliseconds
I0821 19:52:04.844] I0821 19:51:49.629333     102 helpers.go:199] server response object: [{
I0821 19:52:04.844]   "kind": "Status",
I0821 19:52:04.844]   "apiVersion": "v1",
I0821 19:52:04.844]   "metadata": {},
I0821 19:52:04.845]   "status": "Failure",
I0821 19:52:04.845]   "message": "Unauthorized",
I0821 19:52:04.846]   "reason": "Unauthorized",
I0821 19:52:04.846]   "code": 401
I0821 19:52:04.846] }]
I0821 19:52:04.846] F0821 19:51:49.629526     102 helpers.go:114] error: You must be logged in to the server (Unauthorized)
I0821 19:52:04.847]  + /tmp/kubectl get pods '--token=invalid' '--v=7'
I0821 19:52:04.847] command terminated with exit code 255
I0821 19:52:04.848]  [] <nil> 0xc0048fcf90 exit status 255 <nil> <nil> true [0xc00592a1e8 0xc00592a200 0xc00592a218] [0xc00592a1e8 0xc00592a200 0xc00592a218] [0xc00592a1f8 0xc00592a210] [0x9d21f0 0x9d21f0] 0xc001ba2360 <nil>}:
I0821 19:52:04.848] Command stdout:
```

**Comment by odinuge** [comment]
Hi,
Looks like this is a result of https://github.com/kubernetes/kubernetes/pull/81330. The round tripper will now mask the token, replacing it with `<masked>` in the logs. The test (pre that PR) looked for the exact match of the token, in this case the value `invalid`.

The line in the test that fails the test is removed from master, but it is still a part of all current releases, except from `v1.16.0-beta.1`.

**Comment by alejandrox1** [comment]
Thank you @odinuge for the quick response!
I was looking over the test code in master puzzled why i saw no `gomega.Expect(err).To(gomega.ContainSubstring("Authorization: Bearer invalid"))` :sweat_smile:

**Comment by liggitt** [comment]
> The test (pre that PR) looked for the exact match of the token, in this case the value `invalid`.
> 
> The line in the test that fails the test is removed from master, but it is still a part of all current releases, except from `v1.16.0-beta.1`.

whatever branches contain tests that run against 1.16 servers must remove the line that expects the token to be logged

**Comment by odinuge** [comment]
Do you want me to create patches removing the line for the `v1.15` and `v1.14` branches (looks like those run the tests) @alejandrox1?

**Comment by liggitt** [comment]
> Do you want me to create patches removing the line for the `v1.15` and `v1.14` branches (looks like those run the tests) @alejandrox1?

we run 1.14 e2e tests against master? that's unexpected

**Comment by alejandrox1** [comment]
At least for 1.15. I haven't seen that error show up anywhere else

**Comment by odinuge** [comment]
> we run 1.14 e2e tests against master? that's unexpected

You are correct, only 1.15. Must have read something wrong :sweat_smile:

**Comment by odinuge** [comment]
I just thought i would let you know that the PR was merged into the `release-1.15` branch today, and the first two tests just passed! :smile: 

https://testgrid.k8s.io/sig-cli-master#skew-cluster-stable1-kubectl-latest-gce

@alenkacz

**Comment by alejandrox1** [comment]
Thank you for the update @odinuge 
/close
